### PR TITLE
esp32 flashing

### PR DESCRIPTION
**Pull request by jlpoltrack**
To address https://github.com/olliw42/mLRS-docu/issues/158

### CODE REVIEW

**Comment by olliw42** [comment]
just for my curiosity
why doesn't there seem to be such hurdles when flashing using e.g ArduinoIDE, or VsCode

**Comment by jlpoltrack** [comment]
At least for VSCode, it does this transparently (can be observed when you do a verbose upload)

**Comment by olliw42** [comment]
so, you're sayin that vscode uses esptools in exatly that way, to flash all these files ?

**Comment by jlpoltrack** [comment]
> so, you're sayin that vscode uses esptools in exatly that way, to flash all these files ?

Yea, that's where I got these files + offsets: 

image

**Comment by olliw42** [comment]
wouldn't it maybe easier to just provide a py script which a user can call
we could copy these files and esptool into a subfolder in the tools menu

**Comment by olliw42** [comment]
@jlpoltrack besdies my questiions, I think this should go in rather quickly and not later

just two comments:
- maybe a screenshoot also for the ESP8285 case could be useful
- download the zip file isn't very convenient, since on is directed to a raw github page. Not sure, is it possible to provide a link which starts the download? I guess one also could mentgion that it's a zip file, so also need unpacking

**Comment by jlpoltrack** [comment]
- ESP8266 screenshot added
- Zip link will now download the file directly
- Added in the iFlight targets as well (since the edits affect the same page)

File: docs/ELRS_RECEIVERS.md [hunk 1]
```diff
@@ -54,6 +56,15 @@ Note: In order to send RC channels over the serial connection, change the Rx Snd
     - For receivers with a single (non-RGB) LED you can confirm the receiver is in bootloader mode if the LED is solid.
 - Open a supported browser (Chrome, Edge, Opera, not Firefox) and navigate [here](https://esp.huhn.me/).
 - Click connect and select the serial port of your USB<>UART.
-- Upload the .bin file for your receiver and click program.
+- For ESP8266 based receivers:
+    - Upload the .bin file for your receiver:
+    - <img src="images/ESP8266_Flash.png">
```
Comment (olliw42) [inline docs/ELRS_RECEIVERS.md:61]: for some reasons my browser refuses to show it, probably just a problem on my end

note: the "-" does create a dangling o ...  maybe another method of insertuingthe fig would be better, but just nitpick

File: docs/ELRS_RECEIVERS.md [hunk 2]
```diff
@@ -54,6 +56,15 @@ Note: In order to send RC channels over the serial connection, change the Rx Snd
...
+    - Click program.
+- For ESP32 based receivers:
+    - Download the additional files needed for flashing [here](https://github.com/olliw42/mLRS-docu/raw/esp32-flashing/docs/files/esp32_files.zip).
+    - Create additional entries with the following offsets and files.  
+    - Note: the firmware.bin should use the 10000 offset.
+    - <img src="images/ESP32_Flash.png">
```
Comment (olliw42) [inline docs/ELRS_RECEIVERS.md:67]: note: the "-" does create a dangling o ...  maybe another method of insertuingthe fig would be better, but just nitpick

File: docs/ELRS_RECEIVERS.md [hunk 3]
```diff
@@ -54,6 +56,15 @@ Note: In order to send RC channels over the serial connection, change the Rx Snd
     - For receivers with a single (non-RGB) LED you can confirm the receiver is in bootloader mode if the LED is solid.
 - Open a supported browser (Chrome, Edge, Opera, not Firefox) and navigate [here](https://esp.huhn.me/).
 - Click connect and select the serial port of your USB<>UART.
-- Upload the .bin file for your receiver and click program.
+- For ESP8266 based receivers:
+    - Upload the .bin file for your receiver:
+    - <img src="images/ESP8266_Flash.png">
+    - Click program.
+- For ESP32 based receivers:
+    - Download the additional files needed for flashing [here](https://github.com/olliw42/mLRS-docu/raw/esp32-flashing/docs/files/esp32_files.zip).
```
Comment (olliw42) [inline docs/ELRS_RECEIVERS.md:64]: maybe add " (do not forget to unpack the zip file)"

File: docs/ELRS_RECEIVERS.md (hunk 1 above)
Comment (jlpoltrack) [reply docs/ELRS_RECEIVERS.md:61]: Added in breaks instead, looks cleaner.

File: docs/ELRS_RECEIVERS.md (hunk 2 above)
Comment (jlpoltrack) [reply docs/ELRS_RECEIVERS.md:67]: Added in breaks instead, looks cleaner.

File: docs/ELRS_RECEIVERS.md (hunk 3 above)
Comment (jlpoltrack) [reply docs/ELRS_RECEIVERS.md:64]: Added this to the step.

**Comment by olliw42** [comment]
except that the esp8285 won't be shown for me, it lgtm :)

THX!!!

**Comment by olliw42** [comment]
ahh ... maybe it needs to be equal case, so the file needs to be renamed to ESP8266_Flash.png

**Comment by jlpoltrack** [comment]
> ahh ... maybe it needs to be equal case, so the file needs to be renamed to ESP8266_Flash.png

Yea, the VSCode markdown parser didn't mind this - but fixed now.

**Comment by olliw42** [comment]
now it shows
gorgeous

lgtm

MANY THX !!!

**Pull request merged**